extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -56,11 +56,48 @@ public static function onBeforeInitialize( Title $title, $unused, OutputPage $ou
 				!self::httpsEnabledTitle( $title )
 			) {
 				$output->redirect( wfHttpsToHttp( $requestURL ) );
+				$output->enableClientCache( false );
 			}
 		}
 		return true;
 	}
 
+	/**
+	 * Handle downgrading anonymous requests for our sitemaps.
+	 *
+	 * @param  string     $subpage Specific sitemap being requested.
+	 * @param  WebRequest $request
+	 * @param  User       $user
+	 * @return boolean
+	 */
+	public static function onSitemapPageBeforeOutput( string $subpage, WebRequest $request, User $user ): bool {
+		global $wgScriptPath;
+		if ( WebRequest::detectProtocol() === 'https' &&
+			!self::httpsAllowed( $user, $request->getFullRequestURL() )
+		) {
+			self::downgradeRedirectForPath( "$wgScriptPath/$subpage", $request );
```

Comment:
So we're undoing the rewrite introduced in VCL. Assuming the rewrite actually happened. And skipping all the query params (like a `cb=`). It should be fine in this case, but makes me wonder if we should have any rewrites for app on VCL level once we're doing protocol redirects in the app.

Reply:
as discussed, we will do some cleanup in a follow-up ticket